engine/src/ui/color_rect/plugin.cpp

```diff
@@ -88,28 +84,18 @@ void cubos::engine::colorRectPlugin(Cubos& cubos)
         .tagged(uiDrawTag)
         .with<UIElement>()
         .with<UIColorRect>()
-        .call([](const State& state, const Window& window, Query<UIElement&, const UIColorRect&> query) {
-            window->renderDevice().setDepthStencilState(state.dss);
-            window->renderDevice().setViewport(0, 0, static_cast<int>(window->framebufferSize()[0]),
-                                               static_cast<int>(window->framebufferSize()[1]));
-            window->renderDevice().setVertexArray(state.va);
-            window->renderDevice().setShaderPipeline(state.pipeline);
+        .call([](State& state, Query<UIElement&, const UIColorRect&> query) {
             for (auto [element, colorRect] : query)
             {
                 float minX = element.position.x - element.size.x * (element.pivot.x);
                 float minY = element.position.y - element.size.y * (element.pivot.y);
                 float maxX = element.position.x + element.size.x * (1.0F - element.pivot.x);
                 float maxY = element.position.y + element.size.y * (1.0F - element.pivot.y);
 
-                PerElement perElement{{minX, maxX}, {minY, maxY}, element.layer * 100 + element.hierarchyDepth};
-                state.perElement->fill(&perElement, sizeof(perElement));
-                state.pipeline->getBindingPoint("PerElement")->bind(state.perElement);
-
-                auto mvpBuffer = window->renderDevice().createConstantBuffer(sizeof(glm::mat4), &element.vp,
-                                                                             cubos::core::gl::Usage::Static);
-                state.pipeline->getBindingPoint("MVP")->bind(mvpBuffer);
-                state.color->setConstant(colorRect.color);
-                window->renderDevice().drawTriangles(0, 6);
+                element.draw<PerElement>(
```

Comment:
```suggestion
                element.draw(
```
Isn't the compiler able to infer the type?